numpy/random/_bounded_integers.pyx.in

```diff
@@ -314,10 +314,9 @@ cdef object _rand_{{nptype}}(object low, object high, object size,
     high_arr = <np.ndarray>np.array(high, copy=False)
     low_ndim = np.PyArray_NDIM(low_arr)
     high_ndim = np.PyArray_NDIM(high_arr)
-    if ((low_ndim == 0 or (low_ndim == 1 and low_arr.size == 1 and size is not None)) and
-            (high_ndim == 0 or (high_ndim == 1 and high_arr.size == 1 and size is not None))):
-        low = int(low_arr)
-        high = int(high_arr)
+    if low_ndim == 0 and high_ndim == 0:
+        low = low_arr
+        high = high_arr
```

Comment:
No cast to `int` needed any more?

Reply:
Yeah, tests are failing because of this. I'd recommend not opening the can of worms that is working out why, and just put the `int` back for now.

Reply:
Ah yes, I've seen float limits before too. That's fine.

Reply:
Floats seem like they ought to be forbidden here - perhaps a follow-up should switch this to `operator.index`